administrative-sdk/connection.js

```diff
@@ -229,46 +232,44 @@ class Connection {
    * @param {callback} [ecb] The callback that handles the error response.
    * @param {string} [auth] The authorization header value to pass along with the request.
    */
-  _ajaxPost(url, formdata, cb, ecb, auth) {
-    var self = this;
-    var request = new XMLHttpRequest();
+  _ajaxPost(url, formdata, auth) {
+    var request = new XMLHttpRequest({mozSystem: true});
```

Comment:
Part of converting to promises is to rewrite XMLHttpRequest to the new fetch API.